python/cuml/tests/test_target_encoder.py

```diff
@@ -277,3 +277,19 @@ def test_get_params():
     p2 = encoder.get_params()
     for k, v in params.items():
         assert v == p2[k]
+
+
+def test_targetencoder_median():
```

Comment:
Test is fine, though building from my comment about separating the logic of calculating things from the init, we could have lower level stat tests, what do you think?